### DISCUSSION TITLE
Custom Shader Integration in "model-viewer"

### DISCUSSION THREAD

**Discussion by ghost** [google/model-viewer#4677] [Ideas]
Dear Google Viewer Team,

I am currently using your "model-viewer" for displaying jewelry products and have noticed that the existing materials API does not fully capture the realistic look of diamonds and jewelry. Integrating custom shader capabilities would greatly enhance the visual quality of these items.

Could you please consider adding custom shader support to "model-viewer"? Information about any upcoming plans for such a feature would be highly valued.

Thank you for your time and consideration.
briodev

**Comment by elalish** [comment]
We have seen some impressive custom shaders for gem rendering, but I'm not sure if any are open source. Also, they tend to be fairly heavy, requiring a decent GPU to hit framerate, while here we try to enable a smooth experience across the widest range of devices. We'd certainly be open to PRs if anyone has a good solution they want to share.
